src/py/flwr/server/driver/grpc_driver.py

```diff
@@ -46,92 +46,94 @@
 ERROR_MESSAGE_DRIVER_NOT_CONNECTED = """
 [Driver] Error: Not connected.
 
-Call `connect()` on the `GrpcDriverHelper` instance before calling any of the other
-`GrpcDriverHelper` methods.
+Call `connect()` on the `GrpcDriverStub` instance before calling any of the other
+`GrpcDriverStub` methods.
 """
 
 
-class GrpcDriverHelper:
-    """`GrpcDriverHelper` provides access to the gRPC Driver API/service."""
+class GrpcDriverStub(DriverStub):
+    """`GrpcDriverStub` provides access to the gRPC Driver API/service."""
 
     def __init__(
         self,
         driver_service_address: str = DEFAULT_SERVER_ADDRESS_DRIVER,
         root_certificates: Optional[bytes] = None,
     ) -> None:
+        event(EventType.DRIVER_CONNECT)
         self.driver_service_address = driver_service_address
         self.root_certificates = root_certificates
-        self.channel: Optional[grpc.Channel] = None
-        self.stub: Optional[DriverStub] = None
-
-    def connect(self) -> None:
-        """Connect to the Driver API."""
-        event(EventType.DRIVER_CONNECT)
-        if self.channel is not None or self.stub is not None:
-            log(WARNING, "Already connected")
-            return
         self.channel = create_channel(
             server_address=self.driver_service_address,
             insecure=(self.root_certificates is None),
             root_certificates=self.root_certificates,
         )
-        self.stub = DriverStub(self.channel)
+        super().__init__(self.channel)
         log(DEBUG, "[Driver] Connected to %s", self.driver_service_address)
 
     def disconnect(self) -> None:
         """Disconnect from the Driver API."""
         event(EventType.DRIVER_DISCONNECT)
-        if self.channel is None or self.stub is None:
+        if self.channel is None:
             log(DEBUG, "Already disconnected")
             return
         channel = self.channel
         self.channel = None
-        self.stub = None
         channel.close()
         log(DEBUG, "[Driver] Disconnected")
 
     def create_run(self, req: CreateRunRequest) -> CreateRunResponse:
         """Request for run ID."""
         # Check if channel is open
-        if self.stub is None:
+        if self.channel is None:
             log(ERROR, ERROR_MESSAGE_DRIVER_NOT_CONNECTED)
-            raise ConnectionError("`GrpcDriverHelper` instance not connected")
+            raise ConnectionError("`GrpcDriverStub` instance not connected")
 
         # Call Driver API
-        res: CreateRunResponse = self.stub.CreateRun(request=req)
+        res: CreateRunResponse = self.CreateRun(request=req)
```

Comment:
The subclassing leads creates a `GrpcDriverStub` that exposes both `create_run` and `CreateRun` as methods. Would it perhaps be better to keep the internal `stub` object to have only `create_run`? One alternative - perhaps for a different PR - would be to get rid of `GrpcDriverStub`/`GrpcDriverHelper` altogether and just give `GrpcDriver` and internal `stub` object (without an additional layer of indirection)

Reply:
Ok, sure. I've reverted the changes.